### ISSUE TITLE
model.update() doesn't handle affectedRows == null

### ISSUE THREAD

**Issue by mariusa** [sequelize/sequelize#14157]
https://github.com/sequelize/sequelize/blob/ac6bc26fc2e40fc0b0d9aa9c27195e621d7f753b/src/model.js#L3388

On an update I get the useless error
```
ERROR unhandledRejection Cannot read properties of null (reading 'length')
TypeError: Cannot read properties of null (reading 'length')
    at Function.update (/node_modules/sequelize/lib/model.js:1945:32)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Using postgres, so affectedRows always exists. I'm assuming the issue is with the update fields, but there's no db error shown.
Sequelize should check if `affectedRows == null` and show update details/errors. Help users understand why it failed.

**Comment by ephys** [comment]
Do you have an example query that causes this crash? It would help us narrow it down :)

**Comment by mariusa** [comment]
Sure. I see it happens only when both `returning` & `plain` are set, when there are no updated rows:

``
 await person.update({ phone: '123' }, { where: { email: 'a@a.com' }, returning: true, plain: true })
```

**Comment by njavilas2015** [comment]
Hello, I have the same problem, it does not recognize the correct types.
ideal
![image](https://user-images.githubusercontent.com/47610140/155755771-a73ffb9d-13d2-400c-9bf5-ebeb1ff5d15c.png)

current 

![image](https://user-images.githubusercontent.com/47610140/155755858-732ffb71-0bea-4151-bb2d-784ad465923b.png)


Last usable version without this bug is 6.15.1

**Comment by mariusa** [comment]
@ephys Hi Zoe, is the example clear? sequelize breaks the app everytime there aren't rows to be updated. It's fine to execute update queries where there are no matches, app flow should continue without even a warning.

**Comment by ephys** [comment]
Yep! Thank you. Now it's just a question of trying to reproduce and fixing

**Comment by mariusa** [comment]
Hmm, it seems to happen with `returning: false` too, if there are no updated rows.

**Comment by mariusa** [comment]
I think it also happens here:
https://github.com/sequelize/sequelize/blob/ac6bc26fc2e40fc0b0d9aa9c27195e621d7f753b/src/model.js#L1926

`!Array.isArray(null)` is true

@ephys Would it be possible to please change this check? 
Will this also be fixed in v7 ? 

Thanks
